Chap_API_Sync_Access.tex

```diff
@@ -1,171 +1,422 @@
+\argin{val}{Value to store (handle)}
 \end{arglist}
 
-\returnsimplenb
-
-\returnstart
-\begin{itemize}
-    \item \refconst{PMIX_OPERATION_SUCCEEDED}, indicating that the request was immediately processed and returned \textit{success} - the \refarg{cbfunc} will \textit{not} be called. This can occur if the collective involved only processes on the local node.
-\end{itemize}
-\returnend
+Returns \refconst{PMIX_SUCCESS} or a negative value corresponding to a PMIx error constant.
+If a reserved key is provided in the \refarg{key} argument then \refapi{PMIx_Store_internal} will return \refconst{PMIX_ERR_BAD_PARAM}.
 
-\reqattrstart
-The following attributes are required to be supported by all \ac{PMIx} libraries:
+%%%%
+\descr
 
-\pasteAttributeItem{PMIX_COLLECT_DATA}
-\pasteAttributeItem{PMIX_COLLECT_GENERATED_JOB_INFO}
+Store some data locally for retrieval by other areas of the process.
+This is data that has only internal scope - it will never be posted externally. Typically used to cache data obtained by means outside of \ac{PMIx} so that it can be accessed by various areas of the process.
 
-\reqattrend
 
-\optattrstart
-The following attributes are optional for \ac{PMIx} implementations:
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\subsection{\code{PMIx_Commit}}
+\declareapi{PMIx_Commit}
 
-\pasteAttributeItem{PMIX_ALL_CLONES_PARTICIPATE}
+%%%%
+\summary
 
+Make available to other processes all key-value pairs previously staged via \refapi{PMIx_Put}.
 
-The following attributes are optional for host environments that support this operation:
+%%%%
+\format
 
-\pasteAttributeItem{PMIX_TIMEOUT}
+\versionMarker{1.0}
+\cspecificstart
+\begin{codepar}
+pmix_status_t PMIx_Commit(void);
+\end{codepar}
+\cspecificend
 
-\optattrend
+Returns \refconst{PMIX_SUCCESS} or a negative value corresponding to a PMIx error constant.
 
 %%%%
 \descr
 
-Nonblocking version of the \refapi{PMIx_Fence} routine. See the \refapi{PMIx_Fence} description for further details.
+\ac{PMIx} implementations may choose to locally cache non-reserved keys prior to submitting them for distribution.  \refapi{PMIx_Commit} initiates the operation of making available previously staged key-value pairs to other processes.  Depending on the implementation, this may involve transmitting the entire collection of data posted by the process to a server.  \refapi{PMIx_Commit} is an asynchronous operation that will immediately return to the caller while the data is staged in the background.
 
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-\subsection{Fence-related attributes}
+\adviceuserstart
+Users are advised to always include the call to \refapi{PMIx_Commit} in case the local implementation requires it. Note that posted data will not be circulated during \refapi{PMIx_Commit}. Availability of the data by other processes upon completion of \refapi{PMIx_Commit} therefore still relies upon the exchange mechanisms described at the beginning of this chapter.
```

Comment:
The sentence before this advice contradicts the advice. I think we want to allow the implementation to decide what action to take (or not) behind the scenes - so the text in the prior paragraph is correct. I'd suggest updating this advice with the following:
```suggestion
Users are advised to always include the call to \refapi{PMIx_Commit} in case the local implementation requires it. Note that posted data will not necessarily be circulated during \refapi{PMIx_Commit}. Availability of the data by other processes upon completion of \refapi{PMIx_Commit} therefore still relies upon the exchange mechanisms described at the beginning of this chapter.
```

Reply:
done